pkgs/native_assets_cli/lib/src/api/build_output.dart

```diff
@@ -4,48 +4,123 @@
 
 import 'dart:io';
 
+import 'package:collection/collection.dart';
 import 'package:pub_semver/pub_semver.dart';
+import 'package:yaml/yaml.dart';
 
-import '../model/asset.dart' as model;
-import '../model/build_output.dart' as model;
-import '../model/dependencies.dart' as model;
-import '../model/metadata.dart' as model;
+import '../model/dependencies.dart';
+import '../model/metadata.dart';
+import '../utils/datetime.dart';
+import '../utils/file.dart';
+import '../utils/map.dart';
+import '../utils/yaml.dart';
+import 'architecture.dart';
 import 'asset.dart';
-import 'dependencies.dart';
-import 'metadata.dart';
+import 'build_config.dart';
+import 'os.dart';
 
-abstract class BuildOutput {
-  /// Time the build this output belongs to started.
+part '../model/build_output.dart';
+
+/// The output of a `build.dart` invocation.
+///
+/// A package can have a toplevel `build.dart` script. If such a script exists,
+/// it will be automatically run, by the Flutter and Dart SDK tools. The script
+/// is expect to produce a specific output which [BuildOutput] can produce.
+abstract final class BuildOutput {
+  /// Start time for the build of this output.
   ///
-  /// Rounded down to whole seconds, because [File.lastModified] is rounded
-  /// to whole seconds and caching logic compares these timestamps.
+  /// The [timestamp] is rounded down to whole seconds, because
+  /// [File.lastModified] is rounded to whole seconds and caching logic compares
+  /// these timestamps.
   DateTime get timestamp;
-  List<Asset> get assets;
-  Dependencies get dependencies;
-  Metadata get metadata;
 
+  /// The assets produced by this build.
+  ///
+  /// In dry runs, the assets for all [Architecture]s for the [OS] specified in
+  /// the dry run must be provided.
+  Iterable<Asset> get assets;
+
+  /// The files used by this build.
+  ///
+  /// If any of the files in [dependencies] are modified after [timestamp], the
+  /// build will be re-run.
+  Iterable<Uri> get dependencies;
```

Comment:
Is `sources` maybe a better word? I feel like `dependencies` is a bit confusing.

Reply:
Dep (dependency) is terminology from Clang/GCC/Ninja https://ninja-build.org/manual.html#_depfile, https://ninja-build.org/manual.html#ref_dependencies
You would read the depfile output by the C compiler in `build.dart` and pass the entries into `dependencies`.

What do you find confusing about dependencies?

Reply:
I know the term mostly from package dependencies. It's probably fine, but a name like `thingsWhichTriggerARebuild` would be better if there was a nice word for it..